json/src/main/java/org/openjdk/skara/json/JSONArray.java

```diff
@@ -39,6 +39,24 @@ public JSONArray(JSONValue[] array) {
         }
     }
 
+    private void append(JSONValue value) {
+        if (value instanceof JSONArray) {
+            for (var v : value.asArray()) {
+                append(v);
+            }
+        } else {
+            this.values.add(value);
+        }
+    }
+
+    public JSONArray(JSONValue value, JSONValue... values) {
```

Comment:
Related to the comment about `concat` vs `append`, I would prefer that the constructor does `append` and not `concat`. One could imagine a `static` version of `concat` as well, e.g. `public static JSONArray concat(JSONArray... arrays)`.

Reply:
I don't care about a non-flattening constructor. You can add one if you need one. But I can change it to a static factory method instead, if you want. It's probably clearer, given that we can agree on a suitable name. (I'll accept "concat" if you insist, but I'm not persuaded that it's really better...)